src/d/d_wood.cpp

```diff
@@ -4,184 +4,970 @@
+
+// .rodata
+const Attr_c L_attr = { 
+    {{{
+            0x00C8,
+            0x50,
+            0x5DC,
+            0x32,
+            0.6
+        }, {
+            0x00B4,
+            0x1E,
+            0xC80,
+            0xA,
+            0.2,
+        }}, {{
+            0x01f4,
+            0x96,
+            0x4B0,
+            0x96,
+            0.6,
+        }, {
+            0x02BC,
+            0x32,
+            0x898,
+            0x1E,
+            0.2,
+        }}, {{
+            0x0258,
+            0xC8,
+            0x898,
+            0x12C,
+            0.6,
+        }, {
+            0x01BC,
+            0x1E,
+            0xFA0,
+            0x32,
+            0.2,
+        }}, {{
+            0x0258,
+            0xC8,
+            0x1838,
+            0x1F4,
+            0.6,
+        }, {
+            0x01BC,
+            0x1E,
+            0x3E8,
+            0x32,
+            0.2,
+        }}
+    }, 0x14, 0x14, 0x0e, 18.0f, -3.0f, 2.5f, 200, 1.5f, 0.3f, 80.0f, 15.0f, 80.0f, 15.0f, 100.0f, 40.0f, 0x23, 0x80
+};
+} // namespace
+} // namespace dWood
+
+//-----------------------------------------
+// Constants
+//-----------------------------------------
+const s32 L_Room_Max = 64;
+const float l_Ground_check_y_offset = 100.0f;
+const float l_Ground_check_unk0 = 1.0E+9;
+const double l_Ground_check_unk1 = 0.5;
+const double l_Ground_check_unk2 = 3.0;
+const float kGroundHeightBias = 1.0f;
+
+const u32 l_Oba_swood_bDL_SIZE = 0x100; 
+const u32 l_Oba_swood_b_cutDL_SIZE = 0xc0; 
+
+s32 dWood::Anm_c::M_init_num = 0;
+
+//-----------------------------------------
+// Helpers
+//-----------------------------------------
+namespace {
+inline const Attr_c &attr() { return dWood::L_attr; }
+
+inline const AttrSway_c &attr_sway(AttrSway_e swayID, int idx) {
+    return attr().kSways[swayID][idx];
+}
+} // namespace
+
+//-----------------------------------------
+// Classes
+//-----------------------------------------
 
 /* 800BD678-800BD710       .text __ct__Q25dWood5Anm_cFv */
 dWood::Anm_c::Anm_c() {
-    /* Nonmatching */
+    int iVar1;
+    int iVar2;
+
+    MTXIdentity(mModelMtx);
+    MTXIdentity(mTrunkModelMtx);
+    mMode = Mode_Max;
+    mCountdown = 0;
+    mWindDir = 0;
+    mWindPow = 0.0;
+    mPosOffsetY = 0.0;
+    mPosOffsetZ = 0.0;
+    mVelY = 0.0;
+
+    iVar1 = 0;
+    for (u32 i = 0; i < 2; i++) {
+        mPhaseY[iVar1] = 0;
+        mPhaseX[iVar1] = 0;
+        mAmpY[iVar1] = 0;
+        mAmpX[iVar1] = 0;
+        iVar1 = iVar1 + 1;
+    }
```

Comment:
I notice most of the loops have two separate index variables, is there a reason for this? does it not match if you do:
```suggestion
    for (u32 i = 0; i < 2; i++) {
        mPhaseY[i] = 0;
        mPhaseX[i] = 0;
        mAmpY[i] = 0;
        mAmpX[i] = 0;
    }
```

Reply:
Good catch. That's just copy/paste Ghidra detritus. Fixed!